isisd/isis_tlvs.c

```diff
@@ -1340,6 +1342,87 @@ static int pack_item_ext_subtlvs(struct isis_ext_subtlvs *exts,
 				stream_putl(s, lan->sid);
 		}
 	}
+	/* SRv6 End.X SID as per RFC9352 section #8.1 */
+	if (IS_SUBTLV(exts, EXT_SRV6_ENDX_SID)) {
+		struct isis_srv6_endx_sid_subtlv *adj;
+		size_t subtlv_len;
+		size_t subtlv_len_pos;
+
+		for (adj = (struct isis_srv6_endx_sid_subtlv *)
+				   exts->srv6_endx_sid.head;
+		     adj; adj = adj->next) {
+			stream_putc(s, ISIS_SUBTLV_SRV6_ENDX_SID);
+
+			subtlv_len_pos = stream_get_endp(s);
+			stream_putc(s, 0); /* length will be filled later */
+
+			stream_putc(s, adj->flags);
+			stream_putc(s, adj->algorithm);
+			stream_putc(s, adj->weight);
+			stream_putw(s, adj->behavior);
+			stream_put(s, &adj->sid, IPV6_MAX_BYTELEN);
+
+			if (adj->subsubtlvs) {
+				/* Pack Sub-Sub-TLVs */
+				if (isis_pack_subsubtlvs(adj->subsubtlvs, s))
+					return 1;
+			} else {
+				/* No Sub-Sub-TLVs */
+				if (STREAM_WRITEABLE(s) < 1) {
+					*min_len = 21;
```

Comment:
Magic number "21", do we want to define it?

Reply:
Fixed.